src/Estimators/src/SubModelKinDynWrapper.cpp

```diff
@@ -0,0 +1,273 @@
+                              subModel.getAccelerometerList()[idx].frame);
+            return false;
+        }
+
+        // Update dJnu
+        dJnuList[subModel.getAccelerometerList()[idx].name]
+            = kinDyn.getFrameBiasAcc(subModel.getAccelerometerList()[idx].frame);
+
+        // Update rotMatrix
+        iDynTree::Transform H
+            = kinDyn.getWorldTransform(subModel.getAccelerometerList()[idx].frame);
+        accRworldList[subModel.getAccelerometerList()[idx].name] = H.getRotation().inverse();
+    }
+
+    // Update gyroscope jacobians
+    for (int idx = 0; idx < subModel.getGyroscopeList().size(); idx++)
+    {
+        if (!kinDyn.getFrameFreeFloatingJacobian(subModel.getGyroscopeList()[idx].frame,
+                                                 jacGyroList[subModel.getGyroscopeList()[idx].name]))
+        {
+            blf::log()->error("{} Unable to get the compute the free floating jacobian of the "
+                              "frame {}.",
+                              logPrefix,
+                              subModel.getGyroscopeList()[idx].frame);
+            return false;
+        }
+    }
+
+    // Update external contact jacobians
+    for (int idx = 0; idx < subModel.getExternalContactList().size(); idx++)
+    {
+        if (!kinDyn.getFrameFreeFloatingJacobian(subModel.getExternalContactList()[idx],
+                                                 jacContactList[subModel
+                                                                    .getExternalContactList()[idx]]))
+        {
+            blf::log()->error("{} Unable to get the compute the free floating jacobian of the "
+                              "frame {}.",
+                              logPrefix,
+                              subModel.getExternalContactList()[idx]);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool RDE::SubModelKinDynWrapper::inverseDynamics(Eigen::VectorXd& motorTorqueAfterGearbox,
+                                                 Eigen::VectorXd& frictionTorques,
+                                                 Eigen::VectorXd& tauExt,
+                                                 Eigen::VectorXd& baseAcceleration,
+                                                 Eigen::VectorXd& jointAcceleration)
+{
+    //    auto massMatrixEigen(iDynTree::toEigen(massMatrix));
+    long unsigned int nrOfJoints = subModel.getModel().getNrOfDOFs();
+    auto FTEigen(iDynTree::toEigen(massMatrix).block(6, 0, nrOfJoints, 6));
+    auto HEigen(iDynTree::toEigen(massMatrix).block(6, 6, nrOfJoints, nrOfJoints));
+
+    auto genBiasTorquesEigen(iDynTree::toEigen(genForces.jointTorques()));
+
+    auto pseudoInverseHEigen = HEigen.completeOrthogonalDecomposition().pseudoInverse();
+
+    jointAcceleration = pseudoInverseHEigen
+                        * (motorTorqueAfterGearbox - frictionTorques - genBiasTorquesEigen + tauExt
+                           + FTEigen * baseAcceleration);
```

Comment:
This code can generate a lot of dynamically allocated memory, see https://eigen.tuxfamily.org/dox/TopicLazyEvaluation.html. There is a tradeoff between code legibility and performance. If you are more interested in the second, then you should split the expressions every time you have a multiplication, for example
```    
temp1 = FTEigen * baseAcceleration;
temp2 = motorTorqueAfterGearbox - frictionTorques - genBiasTorquesEigen + tauExt + temp1;
jointAcceleration = pseudoInverseHEigen * temp2;
```
with ``temp1`` and ``temp2`` other class members useful to store intermediate results. Clearly, you can choose names better than ``temp1`` and ``temp2`` :wink: